packages/kbn-alerts-ui-shared/src/common/hooks/use_fetch_alerts_fields_query.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { AlertConsumers, isValidFeatureId } from '@kbn/rule-data-utils';
+import { useQuery } from '@tanstack/react-query';
+import type { QueryOptionsOverrides } from '../types/tanstack_query_utility_types';
+import { fetchAlertsFields, FetchAlertsFieldsParams } from '../apis/fetch_alert_fields';
+
+export type UseFetchAlertsFieldsQueryParams = FetchAlertsFieldsParams;
+
+const UNSUPPORTED_FEATURE_ID = AlertConsumers.SIEM;
+
+export const queryKeyPrefix = ['alerts', fetchAlertsFields.name];
+
+/**
+ * Fetch alerts indexes browser fields for the given feature ids
+ *
+ * When testing components that depend on this hook, prefer mocking the {@link fetchAlertsFields} function instead of the hook itself.
+ * @external https://tanstack.com/query/v4/docs/framework/react/guides/testing
+ */
+export const useFetchAlertsFieldsQuery = (
```

Comment:
Good point, I guess that applies more or less to all data fetching hooks? I can see why it was done in [the alerts fetching](https://github.com/elastic/kibana/blob/7f3f757a382f7c567401dde05131beab33b577e0/x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_fetch_alerts.tsx#L269), in this case though the hook is used by multiple other hooks, one of which shows a [combined error message](https://github.com/elastic/kibana/blob/59f5b2398ff84dc40565b2527016d7e6df69750b/packages/kbn-alerts-ui-shared/src/common/hooks/use_alert_data_view.ts#L49) for multiple queries, so perhaps in this case it makes sense to delegate error handling to the caller?

PS We could also decide to show inline prompts instead of toasts by using the `isError` and `error` states returned by RQ depending on the use case.

Reply:
Good points! Ok, let's discuss it in our weekly to decide on a strategy around our fetching hooks.